tests/tls_protected/main.c

```diff
@@ -70,3 +70,9 @@ Test(tls_protected, no_access_lib_secret) {
 Test(tls_protected, access_lib_secret) {
   run_test(true);
 }
+
+#if IA2_REWRITING
+void *__tls_get_addr(size_t m, size_t offset) {
```

Comment:
This could be autogenerated by the rewriter w/o requiring that it appears in user-code, but this was the easiest way to test this. This function is defined in ld.so which is in the same compartment as main.c and the call to `__tls_get_addr` in library.c is required for the rewriter to realize it needs to generate callgates to call it from that compartment. It then adds the `-Wl,--wrap` flag for this function and uses that when linking library.o.